src/Microsoft.ML.Core/Data/IEstimator.cs

```diff
@@ -87,8 +90,13 @@ public bool IsCompatibleWith(Column inputColumn)
                     return false;
                 if (IsKey != inputColumn.IsKey)
                     return false;
-                if (inputColumn.MetadataKinds.Except(MetadataKinds).Any())
-                    return false;
+                foreach (var metaCol in Metadata.Columns)
```

Comment:
@Zruty0 could you double check this, I wrote about [this change here](https://github.com/dotnet/machinelearning/issues/755#issuecomment-419085628); it's a reversal of the earlier logic so I'm not completely confident whether the comment is right or the old code was right, or I somehow got confused. 😄

Reply:
The comment is right, and you fixed it correctly.

---
In reply to: [215618828](https://github.com/dotnet/machinelearning/pull/845#discussion_r215618828) [](ancestors = 215618828)

Reply:
Thanks @Zruty0

---
In reply to: [215722767](https://github.com/dotnet/machinelearning/pull/845#discussion_r215722767) [](ancestors = 215722767,215618828)